src/pages/DoctorDetail.tsx

```diff
@@ -0,0 +1,73 @@
+import { useParams } from "react-router-dom";
+import { MdVerified } from "react-icons/md";
+import { IoIosInformationCircleOutline } from "react-icons/io";
+import { fetchDoctors } from "../components/services/patientService";
+import { specialityImages } from "../assets/assets";
+import useSWR from "swr";
+import { Doctor } from "./ModalDoctorPage";
+
+export function DoctorDetailsFetcher() {
+  const { doctorId } = useParams();
+  const { data, error, isLoading } = useSWR("doctors", fetchDoctors);
+
+  console.log("Doctor ID from URL:", doctorId);
+
+  if (isLoading) return <div>Loading doctor details...</div>;
+  if (error) return <div>An error occurred: {error.message}</div>;
+
+  const doctor = data?.find((doc: { id: string }) => doc.id === doctorId);
+
+  if (!doctor) return <div>No doctor found.</div>;
+
+  return <DoctorDetail doctor={doctor} />;
+}
```

Comment:
Fajnie, domyślam się że jest to kod in progress, przydałyby się tutaj jakieś style no i dedykowany plik dla komponentu.